src/pyapi.jl

```diff
@@ -238,3 +242,39 @@ function write_table(
         description=$description, issues=$issues
     )"
 end
+
+"""
+    download_data_registry(
+        config_filename;
+        data_registry_url=py"DEFAULT_DATA_REGISTRY_URL",
+        token=""
+    )
+
+Download (a subset of) the data registry using the config specified by `config_filename`.
+
+## Keyword arguments
+- `data_registry_url`: the location of the data registry. The default is typically fine.
+- `token`: Registry access token. Shouldn't be needed for downloading.
+"""
+function download_data_registry(
```

Comment:
Is there a way to test this? I'm playing with this locally to understand how to use it and thinking on this too.

Reply:
I'm not sure, we could put in a test where it connects to the online data repository, but that doesn't seem right. Ideally we could make a fake local repository but I don't know how to do that

Reply:
I can't see any coverage for this anyway for data_pipeline_api.

Reply:
Wait, can't we use the `data_config.yaml` as suggested in the MR write up and look for the `metadata.yaml` file downloaded?

Reply:
Yes we could do. Anything changing at data.scrc.uk (or any server outages) would cause the tests to fail though

Reply:
But maybe that's ok?

Reply:
I think that we can test for the appearance of the "human/infection.../0.1.0.toml" file. 

i.e. add this `data_config.yaml`:

"
idata_directory: data
access_log: access-{run_id}.yaml
fail_on_hash_mismatch: True
namespace: SCRC
read:
  - where:
      data_product: human/infection/SARS-CoV-2/asymptomatic-period
      component: asymptomatic-period
"

Then add this to tests:
```
@test !isfile("./data/human/infection/SARS-CoV-2/asymptomatic-period/0.1.0.toml")
download_data_registry("./data/data_config.yaml")
@test isfile("./data/human/infection/SARS-CoV-2/asymptomatic-period/0.1.0.toml")
```

Would this work given your comment? Yes it depends on upstream files...but isn't the point that these are supposed to static?

Reply:
I think given that there is no coverage on the data_pipeline_api side, we can accept this as-is.

Reply:
Cool yeah I would tend to agree. I will add the testing thing as an issue. My main reservation is it could be annoying if the server goes down (like I think they are down for maintenance all next week, so the tests would fail all next week). But maybe that's actually a feature, because it lets us know if something is broken. But can always add later

Reply:
https://github.com/ScottishCovidResponse/SimulationData.jl/issues/25